### DISCUSSION TITLE
Content workflow permissions

### DISCUSSION THREAD

**Discussion by rijkvanzanten** [directus/directus#3950] [Feature Requests]
Allow multiple validation rules in update permissions w/ setting a dedicated scope rule per validation rule. This would allow powerful workflow configuration. For example, it'll allow you to configure a status field to only be allowed to be a given value, based on the current value of the same status field. For example:

```
For interns:


Item Permissions:
status _in draft, under-review
^^ Can only update items that are currently draft or under review


Validation:

scope: status _eq draft
validation: status _in under-review
^^ Is only allowed to change draft items to under-review

scope: status _eq under-review
validation: status _in draft
^^ Is only allowed to change under-review items back to draft
```

**Comment by nickrum** [comment]
Would it be possible to solve this by using the boolean filter rules without introducing any special handling for this use case?

**Reply by rijkvanzanten** [reply]
Boolean filter rules? I'm not sure I know what you mean

**Reply by nickrum** [reply]
Sorry, should have been more specific! I mean the `_and` and `_or` properties within a rule.
Isn't what you described above kind of like multiple boolean "implications" combined with a boolean "and"?

**Reply by rijkvanzanten** [reply]
> Isn't what you described above kind of like multiple boolean "implications" combined with a boolean "and"?

The idea above is that you can specify the validation rules, based on matched "state" of the item. So you want to configure validation to match `status = under-review`, but _only_ if the current status is `published`. That's currently not possible, as validation is only on the whole item, regardless of it's current state

**Reply by nickrum** [reply]
Aah yeah, makes sense. I was confused by the fact that there is a _current_ value and a value a field is _changed to_. I wonder if there would still be a way to query the _current_ value anywhere inside a rule to make this even more general and powerful.

**Comment by log4shell** [comment]
I like the idea, it would be great to have this feature in permission settings for easy configuration. Currently I've implemented this with a custom hook extension that checks the new status on input and throws an error if it's not an allowed transition based on current status.

However, from what I've seen, the permissions needs to be configured for every collection separately, whereas my hook extension is configured to enforce the rules on any collection when the 'status' field is changed. Is there a way to enforce a common permission model on all or a selection of existing collections ensure consistency and to avoid duplicate work?

**Comment by rijkvanzanten** [comment]
Closed in favor of #2975
